main_test.go

```diff
@@ -0,0 +1,377 @@
+// THE SOFTWARE.
+
+// Package section is responsible for splitting a program into its different
+// sections without attempting to parse the contents.s
+package main
+
+import (
+	"bytes"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/pkg/diff"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Arguments struct {
+	args []string
+	fset *token.FileSet
+}
+
+var command = Arguments{
+	args: []string{"-l", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"},
+	fset: token.NewFileSet(),
+}
+
+func TestNewArgParser(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify correct initialization through newArgParser", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		outputStringLint := argParser.FindOptionByLongName("lint").Description
+		outputStringLintShort := argParser.FindOptionByShortName(rune('l')).Description
+		expectedStringLint := "Turn on lint flag to output patch matches to stdout instead of modifying the files"
+		assert.Equal(t, expectedStringLint, outputStringLint)
+		assert.Equal(t, expectedStringLint, outputStringLintShort)
+		assert.Equal(t, false, opts.Linting)
+		assert.Equal(t, []string(nil), opts.Patches)
+
+	})
+
+	t.Run("Verify functionality of newArgParser", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		patchFileName := command.args[2]
+		assert.Equal(t, true, opts.Linting)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+		require.NoError(t, err)
+
+	})
+
+}
+
+func TestLoadPatches(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify correct functionality of LoadPatches", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		require.NoError(t, err)
+		_, err = loadPatches(command.fset, opts, os.Stdin)
+		require.NoError(t, err)
+	})
+	t.Run("Verify correct error gets thrown when loadPatches Failure", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs([]string{"-l", "-p",
+			"testdata/patch/error1.patch", "testdata/test_files/test2.go"})
+		require.NoError(t, err)
+		path := "testdata/patch/error1.patch"
+		_, err = loadPatches(command.fset, opts, os.Stdin)
+		expectedErrorString := "could not read \"" + path + "\": " + "open " + path + ": no such file or directory"
+		require.EqualError(t, err, expectedErrorString)
+	})
+}
+
+func TestApply(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify functionality of Apply when successful", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		filename := "testdata/test_files/lint_example/test2.go"
+		content, err := ioutil.ReadFile(filename)
+		fast, err := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, err := loadPatches(command.fset, opts, os.Stdin)
```

Comment:
Again, we're retesting things we've already tested.

Although I realize we'll need to call loadPatches because hand-constructing the patch structs might be too painful.
But we still do not need to rely on ParseArgs.